DataSetDocker/Dockerfile

```diff
@@ -62,12 +61,38 @@ RUN conda install --quiet --yes "tornado<5.0.0"
 RUN conda install --quiet --yes pip
 RUN pip install py2nb
 
-RUN find /tmp/dataset/python/examples -name '*.py' -exec py2nb {} \; ;  \
-    bash -c "if [ ! -d /home/$NB_USER/work/examples ]; then mkdir /home/$NB_USER/work/examples; fi;" ; \ 
-    bash -c "if [ -d /tmp/dataset/python/examples ] && [ -d /home/$NB_USER/work/examples ]; then find /tmp/dataset/python/examples -name '*.ipynb' -exec mv {} /home/$NB_USER/work/examples \; ; fi;" ; \
-    chmod -R a+rwx /home/$NB_USER/work
+ARG UPD_EX=/usr/local/bin/update_examples.sh
+
+RUN touch $UPD_EX && \
+    chmod a+x $UPD_EX &&\
+    echo "#!/bin/bash" >> $UPD_EX && \
+    echo "cd /tmp/dataset && git checkout master && git pull" >> $UPD_EX && \
+    echo "find /tmp/dataset/python/examples -name '*.py' -exec py2nb {} \; ;" >> $UPD_EX && \
+    echo "if [ ! -d /home/$NB_USER/work/examples ]; then mkdir /home/$NB_USER/work/examples; fi;" >> $UPD_EX && \
+    echo "if [ -d /tmp/dataset/python/examples ] && [ -d /home/$NB_USER/work/examples ]; then find /tmp/dataset/python/examples -name '*.ipynb' -exec mv {} /home/$NB_USER/work/examples \; ; fi;" >> $UPD_EX && \
+    echo "chmod -R a+rwx /home/$NB_USER/work" >> $UPD_EX && \
+    echo "chmod -R a+rwx /tmp/dataset/python/examples" >> $UPD_EX && \
+    echo "chmod -R a+rwx /tmp/dataset/.git" >> $UPD_EX && \
+    echo "chown -R $NB_USER /home/$NB_USER/work" >> $UPD_EX && \
+    echo "chown -R $NB_USER /tmp/dataset/python/examples" >> $UPD_EX && \
+    echo "chown -R $NB_USER /tmp/dataset/.git" >> $UPD_EX && \
+    echo "exit 0" >> $UPD_EX
+
+RUN bash -x $UPD_EX
+
+RUN sed -i "s/jupyter lab/jupyter lab --NotebookApp.token='' --NotebookApp.password=''/" /usr/local/bin/start-notebook.sh
 
 USER $NB_UID
 
+ARG UPD_NB=/home/$NB_USER/update_examples
```

Comment:
I think calling this `make_examples` might be better and more obvious what it is doing.

Reply:
I don't want to argue a lot about this, but it is exactly "update_repo_and_make_examples", but the part exposed to the user only updates the examples, because some examples exist at the moment. It's basically temporary stuff for the demo in the case we decide to add some examples last moment.

Reply:
ok, I'll rename to update_repo_and_make_examples.sh